scripts/copy-tables.py

```diff
@@ -0,0 +1,113 @@
+import argparse
+import re
+
+from clickhouse_driver import Client
+
+SETTINGS = {
+    "load_balancing": "in_order",
+    "replication_alter_partitions_sync": 2,
+    "mutations_sync": 2,
+    "database_atomic_wait_for_drop_and_detach_synchronously": 1,
+}
+
+
+def _get_client(host, port, user, password, database) -> Client:
+    return Client(
+        host=host,
+        port=port,
+        user=user,
+        password=password,
+        database=database,
+        settings=SETTINGS,
+    )
+
+
+def verify_zk_replica_path(client, database, tables) -> None:
+    """
+    Before we copy over table statements from other nodes, we need to make sure
+    that the current zookeeper paths use the macros defined and align with the path
+    that is generated in the create table statement.
+
+    """
+    print("Looking up macros...")
+    ((_, replica), (_, shard)) = client.execute(
+        "SELECT macro, substitution FROM system.macros"
+    )
+    print(f"Found replica: {replica} for shard: {shard}")
+
+    for table in tables:
+        print(f"Verifying zk replica path for table: {table}...")
+        ((replica_path,),) = client.execute(
+            f"SELECT replica_path FROM system.replicas where table = '{table}'"
+        )
+
+        ((curr_create_table_statement,),) = source_client.execute(
+            f"SHOW CREATE TABLE {database}.{table}"
+        )
+        match = re.search("\/clickhouse(.*,)", curr_create_table_statement)
+        create_table_path = match.group(0)[:-2].replace("{shard}", shard)
+
+        built_replica_path = f"{create_table_path}/replicas/{replica}"
+
+        assert (
+            built_replica_path == replica_path
+        ), f"{built_replica_path} should match zk path: {replica_path}"
+        print(f"Zookeeper replica paths verified for table: {table} ! :)")
+
+
+def copy_tables(source_client, new_client, tables, database, dry_run) -> None:
+
+    verify_zk_replica_path(source_client, args.database, tables)
+
+    if not tables:
+        tables = [result[0] for result in source_client.execute("SHOW TABLES").results]
+
+    show_table_statments = {}
+
+    for name in tables:
+        ((curr_create_table_statement,),) = source_client.execute(
+            f"SHOW CREATE TABLE {database}.{name}"
+        )
+        show_table_statments[name] = curr_create_table_statement
+
+    for table_name, statement in show_table_statments.items():
+        print(f"creating {table_name}...")
+        if dry_run:
+            print(f"create table statement: \n {statement}")
+            return
+        new_client.execute(statement)
```

Comment:
Please put this in an else block to make it explicit that the code is only run when dry_run is set to False.

Reply:
We discussed also flipping it so dry-run is the default and you need to pass `--execute` (similar to `kafkactl`) to actually perform the copy.

Reply:
made the default to run without `execute` but also flipped the check to explicitly do 

```python
if execute:
    target_client.execute(statement)

```